### PR TITLE
Limb-B-Gone

### LINKED ISSUE #4120: "Limb Specific" bomb implant gibs.
Checklist before you submit an issue! Feel free to partially or fully delete this in your eventual report.
Didn't find any other reports of this issue, probably because nobody uses it.

So I used the explosive implant to implant my head, I used the "Limb Specific Explosion" hoping it would pop my head off like a pez dispenser but instead it gibbed my whole body.

Yes with the help of Catnip we tried this multiple times, apperantly no matter where you implant a "Limb Specific" bomb it just gibs you, we tried arms, legs and head and all just gib you.

Talked to Catnip about it, confirmed it I also confirmed it because I'm a mod and I'm staff.

### LINKED ISSUE #8345: Explosive implant "localised limb" is a lie
It gibs, completly, a good suprise.
Tested on IPC, limb selected was a leg. Body got completely destroyed.

### PR DESCRIPTION
**Pull request by Doxxmedearly**
Fixes "Localized Limb" actually being localized to the limb instead of gibbing the target. 
Changes the way explo implants work. The prompt coming up while injecting was pretty cumbersome. Now you use it in-hand to set it. A fallback phrase was added in case people don't set it, and it can explode outside of a person, too, but it's basically useless. We're not going to make these into grenades. 
Uplink box now has the implant outside of the implanter, so you don't have to take it out, set it, then cram it in. 
Uplink info updated with how to use it now.
Fixes #8345 
Fixes #4120

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

File: code/game/objects/items/weapons/implants/implant.dm [hunk 1]
```diff
@@ -163,63 +170,98 @@ Implant Specifics:<BR>"}
...
-		t.hotspot_expose(3500,125)
+/obj/item/implant/explosive/proc/small_boom()
+	if(!imp_in)
+		explosion(get_turf(src), -1, 0, 2, 4)
+	if(ishuman(imp_in) && part)
+		//No tearing off these parts since it's pretty much killing. Mangle them. 
+		if(istype(part,/obj/item/organ/external/chest) ||	\
+				istype(part,/obj/item/organ/external/groin) ||	\
+				istype(part,/obj/item/organ/external/head))
+			part.createwound(BRUISE, 60)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/implants/implant.dm]: Maybe add a lot of pain to the limb. This seems extremely minor.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 2]
```diff
@@ -163,63 +170,98 @@ Implant Specifics:<BR>"}
...
+	addtimer(CALLBACK(src, .proc/small_boom), 25)
 
-	if(t)
-		t.hotspot_expose(3500,125)
+/obj/item/implant/explosive/proc/small_boom()
+	if(!imp_in)
+		explosion(get_turf(src), -1, 0, 2, 4)
+	if(ishuman(imp_in) && part)
+		//No tearing off these parts since it's pretty much killing. Mangle them. 
+		if(istype(part,/obj/item/organ/external/chest) ||	\
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/implants/implant.dm]: It'd be better to typecast the organ and then check if vital == TRUE.

Comment (Doxxmedearly) [reply code/game/objects/items/weapons/implants/implant.dm]: ah, cool, didn't know that was a thing. Time to update oldcode

File: code/game/objects/items/weapons/implants/implant.dm (hunk 1 above)
Comment (Doxxmedearly) [reply code/game/objects/items/weapons/implants/implant.dm]: hilarious that creating a wound doesn't add pain. on it.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 3]
```diff
@@ -230,54 +271,48 @@ Implant Specifics:<BR>"}
 	data.z_transfer = z_transfer
 	SSexplosives.queue(data)
 
-/obj/item/implant/explosive/implanted(mob/source as mob)
+/obj/item/implant/explosive/implanted(mob/source)
+	if(!phrase)
+		phrase = fallback_phrase
+	usr.mind.store_memory("\The [src] in [source] can be activated by saying something containing the phrase ''[phrase]'', <B>say [phrase]</B> to attempt to activate.", 0, 0)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/implants/implant.dm]: Does `usr` even work correctly here? Color me surprised if it does.

**Pull request merged**